src/ProjectTemplates/ComponentsWebAssembly.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/Program.cs

```diff
@@ -19,6 +19,15 @@ public static async Task Main(string[] args)
             builder.RootComponents.Add<App>("app");
 
             // use builder.Services to configure application services.
+			builder.Services.AddSingleton<HttpClient>(s =>
+            {
+                // Creating the URI helper needs to wait until the JS Runtime is initialized, so defer it.
+                var navigationManager = s.GetRequiredService<NavigationManager>();
+                return new HttpClient
+                {
+                    BaseAddress = new Uri(navigationManager.BaseUri)
+                };
+            });
```

Comment:
I suspect this is going to be too much non-obvious code to add into `Program.cs` in the default template.

So as a recommendation, how about adding a new class in `src\Components\WebAssembly\WebAssembly\src\Services` like:

```cs
namespace Microsoft.Extensions.DependencyInjection
{
    public static class HttpClientServiceCollectionExtensions
    {
        public static void AddBaseAddressHttpClient(this IServiceCollection serviceCollection)
        {
            serviceCollection.AddSingleton(s =>
            {
                // Creating the URI helper needs to wait until the JS Runtime is initialized, so defer it.
                var navigationManager = s.GetRequiredService<NavigationManager>();
                return new HttpClient
                {
                    BaseAddress = new Uri(navigationManager.BaseUri)
                };
            });
        }
    }
}
```

Then in the project template, we just add `services.AddBaseAddressHttpClient();`.

Reply:
Nice idea. How about `.AddDefaultHttpClient();` though? 
The BaseAddress part would be confusing to me if I didn't know the code, since I just now learned that it means `navigationManager.BaseUri` in this context. Up until now, I would have wondered "So where does it get the `BaseAddress` for the HttpClient?". 
Which is coincidentally literally what I've been wondering since I started working with Blazor, as I'm planning to use CORS and had to find out how to safely change the default `BaseAddress`.
Or put it in the description?

Reply:
Let's start with `AddBaseAddressHttpClient`. We have a few API reviews set up at which point we can sort out the naming.

Reply:
Changed to extension method and applied that at the required spots.